### PR TITLE
contribution of Eclipse Ditto helm chart

### PR DESCRIPTION
**Pull request by thjaeckle**
Based on the Helm rework contribution of Kiwigrid https://github.com/eclipse/ditto/pull/553 
Using a new folder structure as mentioned in #2 

Still to verify if that is working as expected - a review from @axdotl wold be much appreciated :)

### CODE REVIEW

**Comment by axdotl** [comment]
I'll have a look on it later this week. Thanks for your patience!

**Comment by ctron** [comment]
This looks great. I would like to understand one thing before taking a deeper look into this: why is the original author of this? Is this you @thjaeckle?

**Comment by thjaeckle** [comment]
No, that's not so easy to define :D
The main author of the "legacy Helm chart" was @stmaute together with a colleague @ghandim
then there were some adjustments by Microsoft
and the main author of the rewrite was @axdotl 
I just adjusted some parts of the chart before copying it here (e.g. not copying the openAPI files, but using them from GitHub).

**Comment by ctron** [comment]
Ok, then I guess we need a proper CQ for that :grin: … could you get this started?!

**Comment by thjaeckle** [comment]
A CQ for a Helm chart which is super special for Eclipse Ditto, really? :D

File: charts/ditto/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "eclipse-ditto.name" -}}
```
Comment (axdotl) [inline charts/ditto/templates/_helpers.tpl]: Usually the properties are named as the chart, means `ditto` i.s.o. `eclipse-ditto`.

**Comment by ctron** [comment]
> A CQ for a Helm chart which is super special for Eclipse Ditto, really? :D

Yes. It is not a dependency CQ, but project code CQ. It is more than 1000 lines of code, and the source is worth investigating. So I think we do need a project code CQ.

File: charts/ditto/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+## Prerequisites
+
+* Has been tested on Kubernetes 1.11+
+
+## Installing the Chart
+
+To install the chart with the release name `eclipse-ditto`, run the following command:
+
+```bash
+helm install kiwigrid/eclipse-ditto --name eclipse-ditto
```
[the remark this answers is not in the document]
Comment (thjaeckle) [reply charts/ditto/README.md]: Yes, I'm however not yet sure which final name we will use for the "repo"

[the remark this answers is not in the document]
Comment (ctron) [reply charts/ditto/README.md]: Let's discuss that in a separate issue.

**Comment by thjaeckle** [comment]
CQ was approved:

> --- Comment #12 from Shawn Kilpatrick <shawn.kilpatrick@eclipse-foundation.org>  2019-12-04 08:22:03 --- We have completed a high level triage of the attachment based on Parallel IP [1].  You may proceed to check the content into the Eclipse repository.  We will perform a comprehensive analysis at a later date based on IP work queue.
[1] http://www.eclipse.org/projects/handbook/#ip-parallel-ip

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.idea
```
Comment (ctron) [inline .gitignore]: I would prefer to have a list which is more tailored towards the content of this repository. And maybe add some comments of "why". At least on the top level of the repository.

File: NOTICE.md [hunk 4]
```diff
@@ -29,6 +29,11 @@ The project maintains the following source code repositories:
 
 * https://github.com/eclipse/packages
 
+# Copyright holders
+
+* Copyright 2019 Bosch Software Innovations GmbH
```
Comment (ctron) [inline NOTICE.md]: As we are adding this section, I would like to see "Red Hat Inc" there as well :wink:

File: .gitignore (hunk 3 above)
Comment (thjaeckle) [reply .gitignore]: So should I comment each `.gitignore` entry?

Comment (thjaeckle) [reply .gitignore]: done ..

File: NOTICE.md (hunk 4 above)
Comment (thjaeckle) [reply NOTICE.md]: done ..

**Comment by ctron** [comment]
Ok. So my understanding is, that this PR only adds the Ditto Helm chart. But doesn't put this into the "Package Zero"?! If that is the case, then how would be include this into the deployment of "Package Zero"?

**Comment by thjaeckle** [comment]
Correct, IMHO we should adjust the folder structure like discussed in #2 :
- put all charts in `charts` folder
- packages in `packages`
- make the Helm charts in `packages` umbrella packages, like e.g. Shown here: https://github.com/CodeJjang/helm-microservices-example

That way charts may be used in several packages.
I didn't want to do that before discussing and the discussion in #2 never started about the folder structure.

We could do that in a follow up PR moving the Hono package (while also adding copyright headers) and establishing the umbrella package zero chart.

**Comment by ctron** [comment]
> We could do that in a follow up PR moving the Hono package (while also adding copyright headers) and establishing the umbrella package zero chart.

Sounds like a plan. We also need to update the Hono charts to more recent ones.

Ok, so the plan is to merge this and then change the structure? That would be ok for me.

**Pull request merged**